### PR TITLE
storage: use $XDG_CACHE_HOME or ~/.cache (+many more)

### PR DESCRIPTION
**Pull request by lasers**
Addresses https://github.com/ultrabug/py3status/issues/1361

### CODE REVIEW

File: py3status/storage.py [hunk 1]
```diff
@@ -16,13 +17,27 @@ class Storage:
...
-        )
-        storage_path = os.path.join(config_dir, 'py3status.data')
-        self.storage_path = storage_path
+
+        self.config = py3_wrapper.config
+        py3_config = self.config.get('py3_config', {})
+        storage_config = py3_config.get('py3status', {}).get('storage')
+
+        if storage_config:
+            storage_file = os.path.expanduser(storage_config)
```
Comment (lasers) [inline py3status/storage.py]: Do we want to `expandvars` too? ... `storage = '$HOME/.tmp/py3status.data'`

**Comment by ultrabug** [comment]
The bomb flag means that for now we're breaking current cache location, we should handle seamless migration by detecting any file present in config_dir and move it to the new default location

Also, we should make sure in the code that the target directory exists, and create it this is not the case

**Comment by lasers** [comment]
>**ultrabug**... we should handle seamless migration by detecting any file present in config_dir and move it to the new default location

I believe that's done. Review this tho.

>**ultrabug** Also, we should make sure in the code that the target directory exists, and create it this is not the case

I think this is bad behavior for `py3status` to do that. Users can mkdir the directories themselves for explicit storage paths. Otherwise, we're already putting them in good places with nice folks.

Doing `touch /tmp/adsada/asdada.txt`... will produce `No such file or directory` on nonexistent dirs and it's much more simple than `py3status`. We should follow `touch`'s behavior.

**Comment by ultrabug** [comment]
> I believe that's done. Review this tho.

I've permitted myself to implement it and log it more explicitly. Also, it is important at that time to respect any new default set by this very PR instead of forcing migration to .cache ;)

Anyway, that's done and tested, merging

**Pull request merged**